src/app/app.component.ts

```diff
@@ -11,8 +14,17 @@ import { IUserInfo } from './userInfo.model';
   styleUrls: ['./app.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class AppComponent implements OnInit {
-  constructor(private readonly router: Router, private readonly authService: AuthService) {}
+export class AppComponent implements OnInit, OnDestroy {
+  loading: boolean = false;
+
+  private readonly _subscriptions: Subscription[] = [];
+
+  constructor(
+    private readonly router: Router,
+    private readonly authService: AuthService,
+    private readonly loaderService: LoaderService,
+    private readonly changeDetectorRef: ChangeDetectorRef
+  ) {}
 
   ngOnInit(): void {
     const token: string = localStorage.getItem('angularCoursesToken');
```

Comment:
Auth service should be responsible for all actions with token

You can do something like
if (authService.isAuthenticated) { ... }
instead of 
if (token) { ... }